src/http_workers/worker_pool.rs

```diff
@@ -0,0 +1,35 @@
+use std::sync::Arc;
+
+use tokio::sync::{mpsc, Mutex};
+
+use super::{worker::Worker, Msg};
+
+pub struct WorkerPool {
+    workers: Vec<Worker>,
+    sender: mpsc::Sender<Msg>,
+}
+
+impl WorkerPool {
+    pub async fn new(size: usize) -> WorkerPool {
+        let (sender, receiver) = mpsc::channel(size);
```

Comment:
`mpsc` as the name implies is `multiple producer single consumer` it means it's not safe (and wrong) to make `multiple consumer` use the same receiver channel (even by holding the receiver in Arc). Hence that is not the right way to push the job to the workers. 

This is why in th mpsc only the `sender` is clonable. because you can create as many copies of the sender (producer) as you want and then you can send them to all `produers` (with no arc)

Also the design pattern is not the right one for this design. Because no, as you can see here https://github.com/threefoldtech/rmb-rs/pull/23/files#diff-b05d8ee11531ddcecb0767abf0309ec418f1a36045f4b749472c6a1cd5334106R43 
you have to get a job from redis first, then you try to push it to the workers. It means you can lose messages. What if you get a message from redis (storage.prcess() returned something) then all the workers are busy, but before any of them is free to handle the message you crashed, or exited, then this message is lost

Instead, (at least to improve the chances of a message not to get lost) is to implement a PULLING mechanism. It means the system will only pop a message from redis to process if and only if a worker is asking for a job.


Hence how I image it as follows (pseudo code)

```
let (sender, receiver) = mpsc::new() ;
// start workers 
for id in 0..size {
 worker::new(sender.clone()); // this spawns the worker
}

// in distributor 
loop {
   let worker = recevier.recv(); // wait until a worker asks for a job
   let msg = stroage.prcess() ; // get message
   worker.send(Job::Message(msg)) ; // or whatever job
}

// inside the worker
loop {
   let (rx, tx) = oneshot::new(); // create a one shot channel
   self.sender.send(tx); // < we push the  transmitter part on the mpsc channel

   job = rx.recv();; // wait for job
}